app/components/types_de_champ_editor/header_sections_summary_component.rb

```diff
@@ -0,0 +1,16 @@
+class TypesDeChampEditor::HeaderSectionsSummaryComponent < ApplicationComponent
+  def initialize(procedure:, is_private:)
+    @procedure = procedure
+    @is_private = is_private
+  end
+
+  def header_sections
+    @procedure.draft_revision
+      .send(@is_private ? :revision_types_de_champ_private : :revision_types_de_champ_public)
+      .filter { _1.type_de_champ.header_section? }
```

Comment:
```suggestion
    coordinates = if @is_private
      @procedure.draft_revision.revision_types_de_champ_private
    else
      @procedure.draft_revision.revision_types_de_champ_public)
     end
     
     coordinates.filter { _1.type_de_champ.header_section? }
```

proposition plus verbeuse mais sans magie noire 🧙🏿‍♂️